sktime/performance_metrics/tests/test_performance_metrics_forecasting.py

```diff
@@ -2,64 +2,524 @@
+    y_true = Y_TEST_CASES[f"test_case_{n_test_case}"]["test"]
+    y_train = Y_TEST_CASES[f"test_case_{n_test_case}"]["train"]
+    # Testing case of perfect forecaster
+    y_pred = y_true
+    # Also perfect benchmark
+    y_pred_benchmark = y_true
 
+    loss_func_args = inspect.getfullargspec(loss_func).args
+    if "y_train" in loss_func_args:
+        function_loss = loss_func(y_true, y_pred, y_train)
+        class_loss = loss_class.fn(y_true, y_pred, y_train)
+    elif "y_pred_benchmark" in loss_func_args:
+        function_loss = loss_func(y_true, y_pred, y_pred_benchmark)
+        class_loss = loss_class.fn(y_true, y_pred, y_pred_benchmark)
+    else:
+        function_loss = loss_func(y_true, y_pred)
+        class_loss = loss_class.fn(y_true, y_pred)
 
-@pytest.mark.parametrize("test_y", TEST_YS)
-def test_mape_loss(test_y):
-    y_test = test_y
-    y_pred = y_test
-    assert mape_loss(y_test, y_pred) == pytest.approx(0.0)
+    # Assertion for functions
+    assert np.isclose(function_loss, true_loss), " ".join(
+        [
+            f"Loss function {loss_func.__name__} returned {function_loss}",
+            f"loss, but {true_loss} loss expected",
+        ]
+    )
+    # Assertion for classes
+    assert np.isclose(class_loss, true_loss), " ".join(
+        [
+            f"Loss function {loss_class.name} returned {class_loss}",
+            f"loss, but {true_loss} loss expected",
+        ]
+    )
 
-    y_pred = y_test * 1.1
-    assert mape_loss(y_test, y_pred) == pytest.approx(0.1)
 
-    y_pred = y_test * 0.9
-    assert mape_loss(y_test, y_pred) == pytest.approx(0.1)
+@pytest.mark.parametrize("loss_func_name", LOSS_RESULTS.keys())
+@pytest.mark.parametrize("n_test_case", [1, 2, 3])
+def test_univariate_loss_value(n_test_case, loss_func_name):
+    loss_class = LOSS_RESULTS[loss_func_name]["class"]
+    loss_func = LOSS_RESULTS[loss_func_name]["func"]
+    true_loss = LOSS_RESULTS[loss_func_name][f"test_case_{n_test_case}"]
+    y_true = Y_TEST_CASES[f"test_case_{n_test_case}"]["test"]
+    y_train = Y_TEST_CASES[f"test_case_{n_test_case}"]["train"]
+    # Use last value as naive forecast to test function
+    y_pred = np.concatenate([y_train, y_true])[23:35]
+    # Just using this nonsensical approach to generate  benchmark for testing
+    y_pred_benchmark = 0.6 * y_pred
 
-    y_pred = y_test * 1.000001
-    assert mape_loss(y_test, y_pred) == pytest.approx(0.000001)
+    class_attrs = loss_class.get_params()
+    loss_func_args = inspect.getfullargspec(loss_func).args
+    if "y_train" in loss_func_args:
+        if len(class_attrs) > 0:
+            function_loss = loss_func(y_true, y_pred, y_train, **class_attrs)
+            class_loss = loss_class.fn(y_true, y_pred, y_train, **class_attrs)
+        else:
+            function_loss = loss_func(y_true, y_pred, y_train)
+            class_loss = loss_class.fn(y_true, y_pred, y_train)
+    elif "y_pred_benchmark" in loss_func_args:
+        if len(class_attrs) > 0:
```

Comment:
this also seems to be something we can factor out into helper function, e.g. `_generate_loss(loss, y_true, y_pred, y_pred_benchmark, class_attrs)` to avoid repeating the if-clause

Reply:
Same as above. Will be fixed in the PR related to #712.